libbeat/processors/add_locale/add_locale.go

```diff
@@ -0,0 +1,34 @@
+package actions
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+)
+
+type addLocale struct {
+	timezone string
+}
+
+func init() {
+	processors.RegisterPlugin("add_locale", newAddLocale)
+}
+
+func newAddLocale(c common.Config) (processors.Processor, error) {
+	zone, _ := time.Now().Zone()
+
+	l := addLocale{timezone: zone}
+
+	return l, nil
+}
+
+func (l addLocale) Run(event common.MapStr) (common.MapStr, error) {
+	event.Put("beat.timezone", l.timezone)
```

Comment:
Given the requirement to have this field automatically reflect the latest timezone, let's not cache the `timezone` like I requested previous (sorry for the change) and make `Run` always call `time.Now().Zone()`. Based on the benchmark results there wasn't a huge cost to doing this.